### ISSUE TITLE
【 run_mlm.py 】attention_mask will be set to [1,1,...1] with DataCollatorForLanguageModeling 

### ISSUE THREAD

**Issue by xieyuchen13** [huggingface/transformers#9304]
tokenized_datasets has been padded when padding="max_length"
when get a dataloader, we will use DataCollatorForLanguageModeling with tokenizer.pad at first
tokenizer.pad will set attention_mask to all 1 because input_ids have already been padded
so i want to know whether the attention mask meet expectations.

**Comment by patrickvonplaten** [comment]
Sorry, I don't really understand the question here. Could you clarify a bit? In case this is a question about the behavior of `DataCollatorForLanguageModeling`, it would be awesome if you could use the forum: https://discuss.huggingface.co/ .
Otherwise it would be great if you provide a code snippet showcasing the unexpected behavior. Thanks!

**Comment by xieyuchen13** [comment]
in run_mlm.py

first use:
`def tokenize_function(examples):
            # Remove empty lines
            examples["text"] = [line for line in examples["text"] if len(line) > 0 and not line.isspace()]
            return tokenizer(
                examples["text"],
                padding=padding,
                truncation=True,
                max_length=data_args.max_seq_length,
                # We use this option because DataCollatorForLanguageModeling (see below) is more efficient when it
                # receives the `special_tokens_mask`.
                return_special_tokens_mask=True,
            )`

if i set padding="max_length", the inputs will be padded. that means the inputs will already be padded to [1, 4, 5, 6, 2, 0, 0..] and attention_mask will be set to [1, 1, 1, 1, 1, 0, 0...]. so when use DataCollatorForLanguageModeling, the inputs will be padded again (tokenizer.pad). the inputs will not change but the attention_mask will be [1, 1, 1, 1, 1, 1, 1..].
if i set padding="false", tokenizer.pad in DataCollatorForLanguageModeling will not pad the inputs to max_length.

**Comment by patrickvonplaten** [comment]
Hey @xieyuchen13 - I don't think the `attention_mask` will change from [1,1,1,...0,0,0] to [1,1,1,....1,1,1] => could you show me an example that proves otherwise?
